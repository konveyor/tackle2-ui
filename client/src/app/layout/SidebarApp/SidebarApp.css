.perspective {
  padding: var(--pf-v5-c-nav__link--PaddingTop)
    var(--pf-v5-c-nav__link--PaddingRight) var(--pf-v5-global--spacer--lg)
    var(--pf-v5-c-nav__link--PaddingLeft);
  border-bottom: 1px solid var(--pf-v5-c-nav__link--before--BorderColor);
}

.perspective .pf-v5-c-select .pf-v5-c-select__toggle {
  color: var(--pf-v5-global--BackgroundColor--100);
  background-color: var(--pf-v5-global--Color--100);
}

.perspective .pf-v5-c-select .pf-v5-c-select__toggle::before {
  border-color: var(--pf-v5-global--BackgroundColor--dark-400)
    var(--pf-v5-global--BackgroundColor--dark-400)
    var(--pf-v5-c-select__toggle--before--BorderBottomColor)
    var(--pf-v5-global--BackgroundColor--dark-400);
}

.perspective .pf-v5-c-select .pf-v5-c-select__toggle:focus::before,
.perspective .pf-v5-c-select .pf-v5-c-select__toggle:hover::before {
  border-bottom-color: var(--pf-v5-c-select__toggle--before--BorderBottomColor);
}

.perspective .pf-v5-c-select .pf-v5-c-select__toggle:focus::before {
  border-bottom-width: var(
    --pf-v5-c-select__toggle--m-expanded--before--BorderBottomWidth
  );
}
