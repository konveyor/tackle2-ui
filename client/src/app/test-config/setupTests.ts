import "@testing-library/jest-dom";
import { server } from "@mocks/server";

const mockInitialized = false;

jest.mock("@react-keycloak/web", () => {
  const originalModule = jest.requireActual("@react-keycloak/web");
  return {
    ...originalModule,
    useKeycloak: () => [mockInitialized],
  };
});

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "localhost:3000/example/path",
  }),
}));

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

// Uncomment the following to see request logging in the console:

// server.events.on("request:start", (req) => {
//   console.log(`Handling a request to ${req.url.href}`);
// });

// server.events.on("request:match", (req) => {
//   console.log(`Request to ${req.url.href} was matched with a handler`);
// });

// server.events.on("request:unhandled", (req) => {
//   console.warn(`Request to ${req.url.href} was not handled`);
// });
