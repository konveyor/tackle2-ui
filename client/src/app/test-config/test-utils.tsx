import React, { FC, ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { Application, Archetype, Assessment } from "@app/api/models";
import { RenderHookOptions, renderHook } from "@testing-library/react-hooks";

import { createContext, useContext } from "react";

const QueryClientContext = createContext<QueryClient | undefined>(undefined);

const AllTheProviders: FC<{
  children: React.ReactNode;
  queryClient?: QueryClient;
}> = ({ children, queryClient }) => {
  const internalQueryClient =
    queryClient ||
    new QueryClient({
      defaultOptions: {
        queries: {
          retry: false,
          cacheTime: 1000,
        },
      },
    });
  return (
    <QueryClientProvider client={internalQueryClient}>
      {children}
    </QueryClientProvider>
  );
};

export const useQueryClientContext = () => {
  const context = useContext(QueryClientContext);
  if (context === undefined) {
    throw new Error(
      "useQueryClientContext must be used within a QueryClientContext.Provider"
    );
  }
  return context;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook = <TProps, TResult>(
  callback: (props: TProps) => TResult,
  options?: Omit<RenderHookOptions<TProps>, "wrapper"> & {
    queryClient?: QueryClient;
  }
) => {
  const { queryClient, ...rest } = options || {};
  const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => (
    <AllTheProviders queryClient={queryClient}>{children}</AllTheProviders>
  );

  return renderHook(callback, { wrapper: Wrapper as React.FC, ...rest });
};

export * from "@testing-library/react";

export { customRender as render };
export { customRenderHook as renderHook };

export const createMockAssessment = (
  overrides: Partial<Assessment> = {}
): Assessment => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    required: true,
    ...overrides,
  } as Assessment;
};

export const createMockApplication = (overrides: Partial<Application> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Application;
};

export const createMockArchetype = (overrides: Partial<Archetype> = {}) => {
  return {
    id: Math.random(),
    name: "Default name",
    description: "Default description",
    ...overrides,
  } as Archetype;
};
