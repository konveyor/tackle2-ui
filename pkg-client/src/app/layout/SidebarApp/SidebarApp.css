.perspective {
  padding: var(--pf-c-nav__link--PaddingTop) var(--pf-c-nav__link--PaddingRight)
    var(--pf-global--spacer--lg) var(--pf-c-nav__link--PaddingLeft);
  border-bottom: 1px solid var(--pf-c-nav__link--before--BorderColor);
}

.perspective .pf-c-select .pf-c-select__toggle {
  color: var(--pf-global--BackgroundColor--100);
  background-color: var(--pf-global--Color--100);
}

.perspective .pf-c-select .pf-c-select__toggle::before {
  border-color: var(--pf-global--BackgroundColor--dark-400)
    var(--pf-global--BackgroundColor--dark-400)
    var(--pf-c-select__toggle--before--BorderBottomColor)
    var(--pf-global--BackgroundColor--dark-400);
}

.perspective .pf-c-select .pf-c-select__toggle:focus::before,
.perspective .pf-c-select .pf-c-select__toggle:hover::before {
  border-bottom-color: var(--pf-c-select__toggle--before--BorderBottomColor);
}

.perspective .pf-c-select .pf-c-select__toggle:focus::before {
  border-bottom-width: var(
    --pf-c-select__toggle--m-expanded--before--BorderBottomWidth
  );
}
