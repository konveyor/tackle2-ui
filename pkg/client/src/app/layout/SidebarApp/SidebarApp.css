.perspective.pf-c-select {
  --pf-global--Color--100: #fff;
  --pf-global--BackgroundColor--100: #000;
}
