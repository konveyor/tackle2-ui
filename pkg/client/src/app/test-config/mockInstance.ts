import MockAdapter from "axios-mock-adapter";
import axios from "axios";

export default new MockAdapter(axios);
