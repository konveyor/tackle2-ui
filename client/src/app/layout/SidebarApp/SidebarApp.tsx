import React, { useEffect } from "react";
import { NavLink, useHistory, useLocation } from "react-router-dom";
import { useTranslation } from "react-i18next";
import {
  Nav,
  NavItem,
  PageSidebar,
  NavList,
  Select,
  SelectOption,
  SelectVariant,
  NavExpandable,
} from "@patternfly/react-core";

import { Paths } from "@app/Paths";
import { LayoutTheme } from "../LayoutUtils";
import { checkAccess } from "@app/common/rbac-utils";
import keycloak from "@app/keycloak";

import { useLocalStorage } from "@migtools/lib-ui";
import { LocalStorageKey } from "@app/Constants";
import { FEATURES_ENABLED } from "@app/FeatureFlags";
import { OptionWithValue, SimpleSelect } from "@app/shared/components";
import { toOptionLike } from "@app/utils/model-utils";
import "./SidebarApp.css";

export const SidebarApp: React.FC = () => {
  const token = keycloak.tokenParsed || undefined;
  const userRoles = token?.realm_access?.roles || [],
    adminAccess = checkAccess(userRoles, ["tackle-admin"]);

  const { t } = useTranslation();
  const { search } = useLocation();
  const history = useHistory();
  enum PersonaKey {
    ADMINISTRATION = "Administration",
    MIGRATION = "Migration",
  }

  const options = [
    <SelectOption
      key="dev"
      component="button"
      value={PersonaKey.MIGRATION}
      isPlaceholder
    >
      {PersonaKey.MIGRATION}
    </SelectOption>,
    ...(adminAccess
      ? [
          <SelectOption
            key="admin"
            component="button"
            value={PersonaKey.ADMINISTRATION}
          >
            {PersonaKey.ADMINISTRATION}
          </SelectOption>,
        ]
      : []),
  ];

  const personaOptions: OptionWithValue<string>[] = [
    {
      value: PersonaKey.MIGRATION,
      toString: () => PersonaKey.MIGRATION,
    },
    {
      value: PersonaKey.ADMINISTRATION,
      toString: () => PersonaKey.ADMINISTRATION,
    },
  ];

  const [selectedPersona, setSelectedPersona] =
    useLocalStorage<PersonaKey | null>(LocalStorageKey.selectedPersona, null);

  useEffect(() => {
    if (!selectedPersona) {
      history.push("/applications");
      setSelectedPersona(PersonaKey.MIGRATION);
    }
  }, []);

  const Navigation = (
    <Nav id="nav-primary" aria-label="Nav" theme={LayoutTheme}>
      <div className="perspective">
        <SimpleSelect
          toggleId="sidebar-perspective-toggle"
          variant={SelectVariant.single}
          aria-label="Select user perspective"
          id="sidebar-perspective"
          value={
            selectedPersona
              ? toOptionLike(selectedPersona, personaOptions)
              : undefined
          }
          options={personaOptions}
          onChange={(selection) => {
            const selectionValue = selection as OptionWithValue<PersonaKey>;
            setSelectedPersona(selectionValue.value);
            if (selectionValue.value === PersonaKey.ADMINISTRATION) {
              history.push(Paths.general);
            } else {
              history.push(Paths.applications);
            }
          }}
        />
      </div>
      {selectedPersona === PersonaKey.MIGRATION ? (
        <NavList title="Global">
          <NavItem>
            <NavLink
              to={Paths.applications + search}
              activeClassName="pf-m-current"
            >
              {t("sidebar.applicationInventory")}
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink to={Paths.reports + search} activeClassName="pf-m-current">
              {t("sidebar.reports")}
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink to={Paths.controls} activeClassName="pf-m-current">
              {t("sidebar.controls")}
            </NavLink>
          </NavItem>
          {FEATURES_ENABLED.migrationWaves ? (
            <NavItem>
              <NavLink to={Paths.waves} activeClassName="pf-m-current">
                {t("sidebar.waves")}
              </NavLink>
            </NavItem>
          ) : null}
          {FEATURES_ENABLED.dynamicReports ? (
            <>
              <NavItem>
                <NavLink to={Paths.issues} activeClassName="pf-m-current">
                  {t("sidebar.issues")}
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink to={Paths.dependencies} activeClassName="pf-m-current">
                  {t("sidebar.dependencies")}
                </NavLink>
              </NavItem>
            </>
          ) : null}
        </NavList>
      ) : (
        <NavList title="Admin">
          <NavItem>
            <NavLink to={Paths.general} activeClassName="pf-m-current">
              {t("terms.general")}
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink to={Paths.identities} activeClassName="pf-m-current">
              {t("terms.credentials")}
            </NavLink>
          </NavItem>
          <NavExpandable
            title="Repositories"
            srText="SR Link"
            groupId="admin-repos"
            isExpanded
          >
            <NavItem>
              <NavLink
                to={Paths.repositoriesGit}
                activeClassName="pf-m-current"
              >
                Git
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink
                to={Paths.repositoriesSvn}
                activeClassName="pf-m-current"
              >
                Subversion
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink
                to={Paths.repositoriesMvn}
                activeClassName="pf-m-current"
              >
                Maven
              </NavLink>
            </NavItem>
          </NavExpandable>
          <NavItem>
            <NavLink to={Paths.proxies} activeClassName="pf-m-current">
              Proxy
            </NavLink>
          </NavItem>
          <NavItem>
            <NavLink to={Paths.migrationTargets} activeClassName="pf-m-current">
              Custom migration targets
            </NavLink>
          </NavItem>
          {FEATURES_ENABLED.migrationWaves ? (
            <NavExpandable
              title="Issue management"
              srText="SR Link"
              groupId="admin-issue-management"
              isExpanded
            >
              <NavItem>
                <NavLink to={Paths.jira} activeClassName="pf-m-current">
                  Jira
                </NavLink>
              </NavItem>
            </NavExpandable>
          ) : null}
        </NavList>
      )}
    </Nav>
  );

  return <PageSidebar nav={Navigation} theme={LayoutTheme} />;
};
