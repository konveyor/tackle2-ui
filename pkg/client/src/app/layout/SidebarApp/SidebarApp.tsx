import React from "react";
import { NavLink, useHistory, useLocation } from "react-router-dom";
import { useTranslation } from "react-i18next";
import {
  Nav,
  NavItem,
  PageSidebar,
  NavList,
  Select,
  SelectOption,
  SelectVariant,
  NavExpandable,
} from "@patternfly/react-core";
import AdminIcon from "@patternfly/react-icons/dist/esm/icons/cogs-icon";
import DevIcon from "@patternfly/react-icons/dist/esm/icons/code-icon";

import { Paths } from "@app/Paths";
import { LayoutTheme } from "../LayoutUtils";
import "./SidebarApp.css";

export const SidebarApp: React.FC = () => {
  const { t } = useTranslation();
  const { search } = useLocation();
  const history = useHistory();

  const onAdminClick = () => {
    console.log("Admin Selected");
  };

  const onDevClick = () => {
    console.log("Dev Selected");
  };

  const options = [
    <SelectOption
      key="dev"
      component="button"
      onClick={onDevClick}
      value="Developer"
      isPlaceholder
    />,
    <SelectOption
      key="admin"
      component="button"
      onClick={onAdminClick}
      value="Administrator"
    />,
  ];

  const [isOpen, setIsOpen] = React.useState(false);
  const [selected, setSelected] = React.useState("Developer");
  const [isDevIcon, setDevIcon] = React.useState(true);

  const onTogglePerspective = () => {
    setIsOpen(!isOpen);
  };

  const onSelectPerspective = (_, selection) => {
    setSelected(selection);
    setIsOpen(!isOpen);
    if (selection === "Administrator") {
      setDevIcon(false);
      history.push(Paths.identities);
    } else {
      setDevIcon(true);
      history.push(Paths.applications);
    }
  };

  const Navigation2 = (
    <Nav id="nav-primary-simple" theme="dark">
      <NavList id="nav-list-simple">
        {/* {routes.map(
          (route, idx) =>
            route.label && (!route.routes ? renderNavItem(route, idx) : renderNavGroup(route, idx))
        )} */}
      </NavList>
    </Nav>
  );

  const Navigation = (
    <>
      <Select
        toggleIcon={isDevIcon ? <DevIcon /> : <AdminIcon />}
        className="perspective"
        variant={SelectVariant.single}
        aria-label="Select user perspective"
        selections={selected}
        isOpen={isOpen}
        onSelect={onSelectPerspective}
        onToggle={onTogglePerspective}
      >
        {options}
      </Select>
      {selected === "Developer" ? (
        <Nav id="nav-primary" aria-label="Nav" theme={LayoutTheme}>
          <NavList title="Global">
            <NavItem>
              <NavLink
                to={Paths.applications + search}
                activeClassName="pf-m-current"
              >
                {t("sidebar.applicationInventory")}
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink
                to={Paths.reports + search}
                activeClassName="pf-m-current"
              >
                {t("sidebar.reports")}
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink to={Paths.controls} activeClassName="pf-m-current">
                {t("sidebar.controls")}
              </NavLink>
            </NavItem>
          </NavList>
        </Nav>
      ) : (
        <Nav id="nav-admin" aria-label="NavAdmin" theme={LayoutTheme}>
          <NavList title="Admin">
            <NavItem>
              <NavLink to={Paths.identities} activeClassName="pf-m-current">
                {t("terms.credentials")}
              </NavLink>
            </NavItem>
            <NavExpandable
              title="Repositories"
              srText="SR Link"
              groupId="admin-repos"
              isExpanded
            >
              <NavItem>
                <NavLink
                  to={Paths.repositoriesGit}
                  activeClassName="pf-m-current"
                >
                  Git
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink
                  to={Paths.repositoriesSvn}
                  activeClassName="pf-m-current"
                >
                  Subversion
                </NavLink>
              </NavItem>
              <NavItem>
                <NavLink
                  to={Paths.repositoriesMvn}
                  activeClassName="pf-m-current"
                >
                  Maven
                </NavLink>
              </NavItem>
            </NavExpandable>
            <NavItem>
              <NavLink to={Paths.proxies} activeClassName="pf-m-current">
                Proxy
              </NavLink>
            </NavItem>
          </NavList>
        </Nav>
      )}
    </>
  );

  return <PageSidebar nav={Navigation} theme={LayoutTheme} />;
};
