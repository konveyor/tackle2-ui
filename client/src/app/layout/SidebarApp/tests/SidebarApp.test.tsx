import { render } from "@app/test-config/test-utils";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { SidebarApp } from "../SidebarApp";

it("Renders without crashing", () => {
  const wrapper = render(
    <Router>
      <SidebarApp />
    </Router>
  );
  expect(wrapper).toMatchSnapshot();
});
