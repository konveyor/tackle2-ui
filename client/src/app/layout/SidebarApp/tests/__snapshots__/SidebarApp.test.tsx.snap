// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-hidden="false"
        class="pf-v5-c-page__sidebar pf-m-expanded"
        id="page-sidebar"
      >
        <nav
          aria-label="Nav"
          class="pf-v5-c-nav"
          data-ouia-component-id="OUIA-Generated-Nav-1"
          data-ouia-component-type="PF5/Nav"
          data-ouia-safe="true"
          id="nav-primary"
        >
          <div
            class="perspective"
          >
            <button
              aria-expanded="false"
              class="pf-v5-c-menu-toggle"
              style="width: 200px;"
              type="button"
            >
              <span
                class="pf-v5-c-menu-toggle__text"
              />
              <span
                class="pf-v5-c-menu-toggle__controls"
              >
                <span
                  class="pf-v5-c-menu-toggle__toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 320 512"
                    width="1em"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
          <ul
            class="pf-v5-c-nav__list"
            role="list"
            title="Admin"
          >
            <li
              class="pf-v5-c-nav__item"
              data-ouia-component-id="OUIA-Generated-NavItem-1"
              data-ouia-component-type="PF5/NavItem"
              data-ouia-safe="true"
            >
              <a
                class="pf-v5-c-nav__link"
                href="/general"
              >
                terms.general
              </a>
            </li>
            <li
              class="pf-v5-c-nav__item"
              data-ouia-component-id="OUIA-Generated-NavItem-2"
              data-ouia-component-type="PF5/NavItem"
              data-ouia-safe="true"
            >
              <a
                class="pf-v5-c-nav__link"
                href="/identities"
              >
                terms.credentials
              </a>
            </li>
            <li
              class="pf-v5-c-nav__item pf-m-expandable pf-m-expanded"
              data-ouia-component-id="OUIA-Generated-NavExpandable-1"
              data-ouia-component-type="PF5/NavExpandable"
              data-ouia-safe="true"
            >
              <button
                aria-expanded="true"
                class="pf-v5-c-nav__link"
              >
                Repositories
                <span
                  class="pf-v5-c-nav__toggle"
                >
                  <span
                    class="pf-v5-c-nav__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v5-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <section
                aria-labelledby="pf-1688687626358z5hypf2xw7s"
                class="pf-v5-c-nav__subnav"
              >
                <h2
                  class="pf-v5-u-screen-reader"
                  id="pf-1688687626358z5hypf2xw7s"
                >
                  SR Link
                </h2>
                <ul
                  class="pf-v5-c-nav__list"
                  role="list"
                >
                  <li
                    class="pf-v5-c-nav__item"
                    data-ouia-component-id="OUIA-Generated-NavItem-3"
                    data-ouia-component-type="PF5/NavItem"
                    data-ouia-safe="true"
                  >
                    <a
                      class="pf-v5-c-nav__link"
                      href="/repositories/git"
                    >
                      Git
                    </a>
                  </li>
                  <li
                    class="pf-v5-c-nav__item"
                    data-ouia-component-id="OUIA-Generated-NavItem-4"
                    data-ouia-component-type="PF5/NavItem"
                    data-ouia-safe="true"
                  >
                    <a
                      class="pf-v5-c-nav__link"
                      href="/repositories/svn"
                    >
                      Subversion
                    </a>
                  </li>
                  <li
                    class="pf-v5-c-nav__item"
                    data-ouia-component-id="OUIA-Generated-NavItem-5"
                    data-ouia-component-type="PF5/NavItem"
                    data-ouia-safe="true"
                  >
                    <a
                      class="pf-v5-c-nav__link"
                      href="/repositories/maven"
                    >
                      Maven
                    </a>
                  </li>
                </ul>
              </section>
            </li>
            <li
              class="pf-v5-c-nav__item"
              data-ouia-component-id="OUIA-Generated-NavItem-6"
              data-ouia-component-type="PF5/NavItem"
              data-ouia-safe="true"
            >
              <a
                class="pf-v5-c-nav__link"
                href="/proxies"
              >
                Proxy
              </a>
            </li>
            <li
              class="pf-v5-c-nav__item"
              data-ouia-component-id="OUIA-Generated-NavItem-7"
              data-ouia-component-type="PF5/NavItem"
              data-ouia-safe="true"
            >
              <a
                class="pf-v5-c-nav__link"
                href="/migration-targets"
              >
                Custom migration targets
              </a>
            </li>
            <li
              class="pf-v5-c-nav__item pf-m-expandable pf-m-expanded"
              data-ouia-component-id="OUIA-Generated-NavExpandable-2"
              data-ouia-component-type="PF5/NavExpandable"
              data-ouia-safe="true"
            >
              <button
                aria-expanded="true"
                class="pf-v5-c-nav__link"
              >
                Issue management
                <span
                  class="pf-v5-c-nav__toggle"
                >
                  <span
                    class="pf-v5-c-nav__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="pf-v5-svg"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      viewBox="0 0 256 512"
                      width="1em"
                    >
                      <path
                        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <section
                aria-labelledby="pf-1688687626367i6ctpip645p"
                class="pf-v5-c-nav__subnav"
              >
                <h2
                  class="pf-v5-u-screen-reader"
                  id="pf-1688687626367i6ctpip645p"
                >
                  SR Link
                </h2>
                <ul
                  class="pf-v5-c-nav__list"
                  role="list"
                >
                  <li
                    class="pf-v5-c-nav__item"
                    data-ouia-component-id="OUIA-Generated-NavItem-8"
                    data-ouia-component-type="PF5/NavItem"
                    data-ouia-safe="true"
                  >
                    <a
                      class="pf-v5-c-nav__link"
                      href="/jira"
                    >
                      Jira
                    </a>
                  </li>
                </ul>
              </section>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-hidden="false"
      class="pf-v5-c-page__sidebar pf-m-expanded"
      id="page-sidebar"
    >
      <nav
        aria-label="Nav"
        class="pf-v5-c-nav"
        data-ouia-component-id="OUIA-Generated-Nav-1"
        data-ouia-component-type="PF5/Nav"
        data-ouia-safe="true"
        id="nav-primary"
      >
        <div
          class="perspective"
        >
          <button
            aria-expanded="false"
            class="pf-v5-c-menu-toggle"
            style="width: 200px;"
            type="button"
          >
            <span
              class="pf-v5-c-menu-toggle__text"
            />
            <span
              class="pf-v5-c-menu-toggle__controls"
            >
              <span
                class="pf-v5-c-menu-toggle__toggle-icon"
              >
                <svg
                  aria-hidden="true"
                  class="pf-v5-svg"
                  fill="currentColor"
                  height="1em"
                  role="img"
                  viewBox="0 0 320 512"
                  width="1em"
                >
                  <path
                    d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
        <ul
          class="pf-v5-c-nav__list"
          role="list"
          title="Admin"
        >
          <li
            class="pf-v5-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-1"
            data-ouia-component-type="PF5/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v5-c-nav__link"
              href="/general"
            >
              terms.general
            </a>
          </li>
          <li
            class="pf-v5-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-2"
            data-ouia-component-type="PF5/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v5-c-nav__link"
              href="/identities"
            >
              terms.credentials
            </a>
          </li>
          <li
            class="pf-v5-c-nav__item pf-m-expandable pf-m-expanded"
            data-ouia-component-id="OUIA-Generated-NavExpandable-1"
            data-ouia-component-type="PF5/NavExpandable"
            data-ouia-safe="true"
          >
            <button
              aria-expanded="true"
              class="pf-v5-c-nav__link"
            >
              Repositories
              <span
                class="pf-v5-c-nav__toggle"
              >
                <span
                  class="pf-v5-c-nav__toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <section
              aria-labelledby="pf-1688687626358z5hypf2xw7s"
              class="pf-v5-c-nav__subnav"
            >
              <h2
                class="pf-v5-u-screen-reader"
                id="pf-1688687626358z5hypf2xw7s"
              >
                SR Link
              </h2>
              <ul
                class="pf-v5-c-nav__list"
                role="list"
              >
                <li
                  class="pf-v5-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-3"
                  data-ouia-component-type="PF5/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    class="pf-v5-c-nav__link"
                    href="/repositories/git"
                  >
                    Git
                  </a>
                </li>
                <li
                  class="pf-v5-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-4"
                  data-ouia-component-type="PF5/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    class="pf-v5-c-nav__link"
                    href="/repositories/svn"
                  >
                    Subversion
                  </a>
                </li>
                <li
                  class="pf-v5-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-5"
                  data-ouia-component-type="PF5/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    class="pf-v5-c-nav__link"
                    href="/repositories/maven"
                  >
                    Maven
                  </a>
                </li>
              </ul>
            </section>
          </li>
          <li
            class="pf-v5-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-6"
            data-ouia-component-type="PF5/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v5-c-nav__link"
              href="/proxies"
            >
              Proxy
            </a>
          </li>
          <li
            class="pf-v5-c-nav__item"
            data-ouia-component-id="OUIA-Generated-NavItem-7"
            data-ouia-component-type="PF5/NavItem"
            data-ouia-safe="true"
          >
            <a
              class="pf-v5-c-nav__link"
              href="/migration-targets"
            >
              Custom migration targets
            </a>
          </li>
          <li
            class="pf-v5-c-nav__item pf-m-expandable pf-m-expanded"
            data-ouia-component-id="OUIA-Generated-NavExpandable-2"
            data-ouia-component-type="PF5/NavExpandable"
            data-ouia-safe="true"
          >
            <button
              aria-expanded="true"
              class="pf-v5-c-nav__link"
            >
              Issue management
              <span
                class="pf-v5-c-nav__toggle"
              >
                <span
                  class="pf-v5-c-nav__toggle-icon"
                >
                  <svg
                    aria-hidden="true"
                    class="pf-v5-svg"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    viewBox="0 0 256 512"
                    width="1em"
                  >
                    <path
                      d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <section
              aria-labelledby="pf-1688687626367i6ctpip645p"
              class="pf-v5-c-nav__subnav"
            >
              <h2
                class="pf-v5-u-screen-reader"
                id="pf-1688687626367i6ctpip645p"
              >
                SR Link
              </h2>
              <ul
                class="pf-v5-c-nav__list"
                role="list"
              >
                <li
                  class="pf-v5-c-nav__item"
                  data-ouia-component-id="OUIA-Generated-NavItem-8"
                  data-ouia-component-type="PF5/NavItem"
                  data-ouia-safe="true"
                >
                  <a
                    class="pf-v5-c-nav__link"
                    href="/jira"
                  >
                    Jira
                  </a>
                </li>
              </ul>
            </section>
          </li>
        </ul>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
